src/main/java/com/deloitte/elrr/aggregator/rules/ProcessCompleted.java

```diff
@@ -0,0 +1,273 @@
+  }
+
+  /**
+   * @param Activity
+   * @param Person
+   * @param Result
+   * @param LearningResource
+   * @return LearningRccord
+   */
+  public LearningRecord processLearningRecord(
+      Activity activity, Person person, Result result, LearningResource learningResource) {
+
+    // Get LearningRecord
+    LearningRecord learningRecord =
+        learningRecordService.findByPersonIdAndLearningResourceId(
+            person.getId(), learningResource.getId());
+
+    // If LearningRecord doesn't exist
+    if (learningRecord == null) {
+      learningRecord = createLearningRecord(person, learningResource, result);
+
+      // If learningRecord already exists
+    } else {
+      learningRecord = updateLearningRecord(learningRecord, result);
+    }
+
+    return learningRecord;
+  }
+
+  /**
+   * @param Person
+   * @param learningResource
+   * @param Result
+   * @return LearningRecord
+   */
+  public LearningRecord createLearningRecord(
+      Person person, LearningResource learningResource, Result result) {
+
+    log.info("Creating new learning record.");
+    LearningRecord learningRecord = new LearningRecord();
+
+    if (result != null) {
+      learningRecord = setStatus(learningRecord, result);
+    } else {
+      learningRecord.setRecordStatus(LearningStatus.ATTEMPTED);
+    }
+
+    learningRecord.setLearningResource(learningResource);
+    learningRecord.setPerson(person);
+    learningRecordService.save(learningRecord);
+
+    log.info(
+        "Learning record for "
+            + person.getName()
+            + " - "
+            + learningResource.getTitle()
+            + " created.");
+
+    return learningRecord;
+  }
+
+  /**
+   * @param person
+   * @param learningRecord
+   * @param learningResource
+   * @param result
+   * @return LearningRecord
+   */
+  public LearningRecord updateLearningRecord(LearningRecord learningRecord, Result result) {
+    log.info("Update learning record.");
+    learningRecord = setStatus(learningRecord, result);
+    learningRecordService.update(learningRecord);
+    return learningRecord;
+  }
+
+  /**
+   * @param learningRecord
+   * @param result
+   * @return learningRecord
+   */
+  private LearningRecord setStatus(LearningRecord learningRecord, Result result) {
+
+    if (result != null) {
+
+      Boolean success = result.getSuccess();
+      Boolean completed = result.getCompletion();
+
+      // status
+      if (completed && success == null) {
+        learningRecord.setRecordStatus(LearningStatus.COMPLETED);
+      } else if (completed && success) {
+        learningRecord.setRecordStatus(LearningStatus.PASSED);
+      } else if (completed && !success) {
+        learningRecord.setRecordStatus(LearningStatus.FAILED);
+      } else {
+        learningRecord.setRecordStatus(LearningStatus.ATTEMPTED);
+      }
+
+      // grade
+      Score score = result.getScore();
+
+      if (score != null) {
+        learningRecord.setAcademicGrade(score.getRaw().toString());
```

Comment:
scaled is actually better here than raw, however we might need to evaluate adding fields to support a more complex score.

Reply:
ignore this for now.